### PR TITLE
K8SPXC-342 haproxy doc

### PR DESCRIPTION
**Pull request by fiowro**
[K8SPXC-342](https://jira.percona.com/browse/K8SPXC-342)

### CODE REVIEW

File: source/proxysql-conf.rst [hunk 1]
```diff
@@ -1,5 +1,40 @@
...
+Configuring Load Balancing with ProxySQL
+========================================
+
+Percona XtraDB Cluster Operator provides choice of two cluster components to
+carry on load balancing and proxy service: you can use either `HAProxy<https://haproxy.org>`_ or `ProxySQL <https://proxysql.com/>`_.
+You can control which one to use, if any, enabling or disabling via the
+``haproxy.enabled`` and ``proxysql.enabled`` options in the ``deploy/cr.yaml``
+configuration file. 
+
+Use the following command to enable HAProxy:
```
Comment (delgod) [inline source/proxysql-conf.rst:12]:
```suggestion
Use the following command to enable ProxySQL:
```

File: source/haproxy-conf.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  this node is available. If zero node is not available, nodes are selected in
+  descending order of their numbers (eg. ``cluster1-pxc-2``, then
+  ``cluster1-pxc-1``, etc.). This service can be used for both read and write
+  load, or it can also be used just for write load (single writer mode) in
+  setups with split write and read loads.
+
+* ``cluster1-haproxy-replicas`` listening on port 3306 (MySQL).
+  This service selects PXC nodes to serve queries following the Round Robin
+  load balancing algorithm.
+
```
Comment (delgod) [inline source/haproxy-conf.rst:37]:
```suggestion
When the cluster with HAProxy is upgraded, the following steps
 take place. First, reader nodes are upgraded one by one: the Operator waits
 until the upgraded PXC member become synced, and then
 proceeds to upgrade the next member. When the upgrade is finished for all reader
 members, then writer PXC member is finally upgraded.
```

File: source/proxysql-conf.rst [hunk 3]
```diff
@@ -1,5 +1,40 @@
...
+   enabling of both HAProxy and ProxySQL.
+
+The resulting setup will use zero PXC node (``cluster1-pxc-0`` by default)
+as writer. 
+
+When the cluster with ProxySQL is upgraded, the following steps
+take place. First, reader nodes are upgraded one by one: the Operator waits
+until the upgraded node shows up in ProxySQL with online status, and then
+proceeds to upgrade the next node. When the upgrade is finished for all reader
+nodes, then writer PXC node is finally upgraded.
```
Comment (delgod) [inline source/proxysql-conf.rst:31]: please `s/node/PXC member/` in all documents.
reason: when we say "node" -- user can think about K8s node

```suggestion
When the cluster with ProxySQL is upgraded, the following steps
take place. First, reader memebers are upgraded one by one: the Operator waits
until the upgraded node shows up in ProxySQL with online status, and then
proceeds to upgrade the next memeber. When the upgrade is finished for all reader
memeber, then the writer PXC member is finally upgraded.
```

File: source/proxysql-conf.rst [hunk 4]
```diff
@@ -1,5 +1,40 @@
-Configuring ProxySQL
-======================
+.. proxysql-conf::
+
+Configuring Load Balancing with ProxySQL
+========================================
+
+Percona XtraDB Cluster Operator provides choice of two cluster components to
```
Comment (tduzan) [inline source/proxysql-conf.rst:6]: "provides a choice of"

File: source/proxysql-conf.rst [hunk 5]
```diff
@@ -1,5 +1,40 @@
...
+carry on load balancing and proxy service: you can use either `HAProxy<https://haproxy.org>`_ or `ProxySQL <https://proxysql.com/>`_.
```
Comment (tduzan) [inline source/proxysql-conf.rst:7]: "provide a load balancing and proxy service"

File: source/proxysql-conf.rst [hunk 6]
```diff
@@ -1,5 +1,40 @@
...
+You can control which one to use, if any, enabling or disabling via the
```
Comment (tduzan) [inline source/proxysql-conf.rst:8]: "if any, by enabling"

File: source/proxysql-conf.rst [hunk 7]
```diff
@@ -1,5 +1,40 @@
...
+
+   kubectl patch pxc cluster1 --type=merge --patch '{ \
+    "spec": {"haproxy":{ "enabled": false }, \
+    "proxysql":{ "enabled": true } \
+    }}'
+
+.. note:: For obvious reasons the Operator will not allow the simultaneous
+   enabling of both HAProxy and ProxySQL.
+
+The resulting setup will use zero PXC node (``cluster1-pxc-0`` by default)
```
Comment (tduzan) [inline source/proxysql-conf.rst:24]: "will use the number zero PXC node"

File: source/proxysql-conf.rst [hunk 8]
```diff
@@ -1,5 +1,40 @@
...
+as writer.
+
+When the cluster with ProxySQL is upgraded, the following steps
```
Comment (tduzan) [inline source/proxysql-conf.rst:27]: "When a cluster with ProxySQL"

File: source/haproxy-conf.rst [hunk 9]
```diff
@@ -0,0 +1,89 @@
+.. haproxy-conf::
+
+Configuring Load Balancing with HAProxy
+=======================================
+
+Percona XtraDB Cluster Operator provides choice of two cluster components to
```
Comment (tduzan) [inline source/haproxy-conf.rst:6]: "a choice"

File: source/haproxy-conf.rst [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+carry on load balancing and proxy service: you can use either `HAProxy<https://haproxy.org>`_ or `ProxySQL <https://proxysql.com/>`_.
```
Comment (tduzan) [inline source/haproxy-conf.rst:7]: "provide load balancing"

File: source/haproxy-conf.rst [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+You can control which one to use, if any, enabling or disabling via the
```
Comment (tduzan) [inline source/haproxy-conf.rst:8]: "any, by"

File: source/haproxy-conf.rst [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+    "proxysql":{ "enabled": false } \
+    }}'
+
+.. note:: For obvious reasons the Operator will not allow the simultaneous
+   enabling of both HAProxy and ProxySQL.
+
+The resulting HAPproxy setup will contain two services:
+
+* ``cluster1-haproxy`` service listening on ports 3306 (MySQL) and 3309 (proxy).
+  This service is pointing to a zero node (``cluster1-pxc-0``) by default when
```
Comment (tduzan) [inline source/haproxy-conf.rst:27]: "to the number zero node"

File: source/haproxy-conf.rst [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+  this node is available. If zero node is not available, nodes are selected in
```
Comment (tduzan) [inline source/haproxy-conf.rst:28]: "If a zero node"

File: source/haproxy-conf.rst [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+configuration file. 
+
+Use the following command to enable HAProxy:
+
+.. code:: bash
+
+   kubectl patch pxc cluster1 --type=merge --patch '{ \
+    "spec": {"haproxy":{ "enabled": true }, \
+    "proxysql":{ "enabled": false } \
+    }}'
```
Comment (delgod) [inline source/haproxy-conf.rst:19]: please check command with @tplavcic 
```suggestion
   kubectl patch pxc cluster1 --type=merge --patch '{ "spec": {
    "proxysql":{ "enabled": false } 
    "haproxy":{ "enabled": true }, 
    "haproxy":{ "size": 3 },
    "haproxy":{ "image": "percona/percona-xtradb-cluster-operator:1.5.0-haproxy" }
    }}'
```

File: source/proxysql-conf.rst (hunk 7 above)
Comment (delgod) [reply source/proxysql-conf.rst:24]: also please `s/node/member/` in all docs

File: source/proxysql-conf.rst [hunk 15]
```diff
@@ -1,5 +1,40 @@
...
+configuration file. 
+
+Use the following command to enable ProxySQL:
+
+.. code:: bash
+
+   kubectl patch pxc cluster1 --type=merge --patch '{ \
+    "spec": {"haproxy":{ "enabled": false }, \
+    "proxysql":{ "enabled": true } \
+    }}'
```
Comment (delgod) [inline source/proxysql-conf.rst:19]: please check command with @tplavcic 
```suggestion
   kubectl patch pxc cluster1 --type=merge --patch '{ "spec": {
    "haproxy":{ "enabled": false } 
    "proxysql":{ "enabled": true }, 
    "proxysql":{ "size": 3 },
    "proxysql":{ "image": "percona/percona-xtradb-cluster-operator:1.5.0-proxysql" }
    }}'
```

**Pull request merged**